### PR TITLE
Apple GameController support

### LINKED ISSUE #11458: Pads: Ability to connect xbox controller on Mac
## Quick summary
Please briefly describe what feature you would like RPCS3 to have.

## Details

Please describe the feature as accurately as possible.

#### 1. Please describe, what part of RPCS3 would be affected by your feature:
* Gameplay
* Debugging
* UI
* Patches
* Installation
* Github
* CI
* etc.

#### 4. Please provide your system configuration:
* OS
* CPU
* GPU
* Driver version
* etc.

#### Please include.
* Anything else you deem to be important

### LINKED ISSUE #11485: Support for non-sony PS4 controllers on Mac-M1
## Quick summary
Support for non-sony PS4 controllers on Mac (M1 max)

## Details
1. Pad is connected via BT
2. Start the RPCS3
3. Click the "Pad" Configure controls icon
4. Got the console log (attached below)
5. PAD is not detected in the PAD-Preferences window (scr shot attached)
6. NOTE: this controller works without issues with the PCSX2 on that machine

## Console log
[Controller contacted via BT]
E DS4: GetCalibrationData: hid_get_feature_report 0x05 for bluetooth controller failed! result=-1, error=0000000002907250
E DS4: check_add_device: GetCalibrationData failed!

[Controller contacted via USB]
E DS4: GetCalibrationData: hid_get_feature_report 0x05 for bluetooth controller failed! result=0, error=0000000002907250
E DS4: check_add_device: GetCalibrationData failed!
E DS4: GetCalibrationData: hid_get_feature_report 0x05 for bluetooth controller failed! result=0, error=0000000002907250
E DS4: check_add_device: GetCalibrationData failed!

## System Info
* OS: MacOs 12.2
* CPU: M1 Max
* GPU M1 Max
* RAM: 64GB

## Controller Info
CUH-ZCT1U
FCC ID: ZAYRM-NS009 212020B00141268

[BLUETOOTH]
  DUALSHOCK 4 Wireless Controller:
  Address:	A0:5A:5C:DF:42:83
  Connected:	Yes
  Vendor ID:	0x054C
  Product ID:	0x05C4
  Firmware Version:	1.0.0
  Minor Type:	Gamepad
  Services:	0x800020 < HID ACL >

[USB]
  Wireless Controller:
  Product ID:	0x09cc
  Vendor ID:	0x054c  (Sony Corporation)
  Version:	1.00
  Speed:	Up to 12 Mb/s
  Manufacturer:	Sony Interactive Entertainment
  Location ID:	0x01110000 / 4
  Current Available (mA):	500
  Current Required (mA):	500
  Extra Operating Current (mA):	0

### PR DESCRIPTION
**Pull request by MaddTheSane**
This will make RPCS3 use Apple's GameController framework, which would add support for Xbox and MFi (made for iPad/iPhone/iOS) controllers.

closes #11485, closes #11458

### CODE REVIEW

**Comment by illusion0001** [comment]
Missing includes in cmakelist, does this work at all?

**Comment by MaddTheSane** [comment]
It does not work at all yet.

**Comment by oresttokovenko** [comment]
Thanks for working on this!

File: rpcs3/Emu/CMakeLists.txt [hunk 1]
```diff
@@ -63,6 +63,7 @@ target_sources(rpcs3_emu PRIVATE
 if(APPLE)
     target_sources(rpcs3_emu PRIVATE
         ../../darwin/util/sysinfo_darwin.mm
+        ../rpcs3/Input/GameController_pad_handler.mm
```
Comment (Megamouse) [inline rpcs3/Emu/CMakeLists.txt:66]: you should put this into
../../darwin/input/GameController_pad_handler.mm

Comment (MaddTheSane) [reply rpcs3/Emu/CMakeLists.txt:66]: Not with the other RPCS3 classes of a similar type?

Comment (Megamouse) [reply rpcs3/Emu/CMakeLists.txt:66]: .mm files belong in the darwin folder

Comment (MaddTheSane) [reply rpcs3/Emu/CMakeLists.txt:66]: Will do.

**Comment by KevDoy** [comment]
Happy that this is being worked on. Thank you.

**Comment by oresttokovenko** [comment]
Hey, any updates?

**Comment by MaddTheSane** [comment]
Sadly, no.

**Comment by RipleyTom** [comment]
I don't want to be the one to do this but someone has to:
-You opened this with nothing to show for it, please have at least a proof of concept before opening a PR.
-You're trying to use an Objective C library in a C++ project, that's not possible(you can use C/C++ libraries in ObjC but not the opposite).

**Comment by MaddTheSane** [comment]
Okay, don't open a pull request, even a draft, without a lot of work. Got it.

**Comment by MaddTheSane** [comment]
Besides, there's something called Objective-C++, which bridges Obj-C and C++.

**Pull request closed without merging**